enhancements/sig-policy/128-resource-selector/README.md

```diff
@@ -0,0 +1,142 @@
+provided. The resource selector would also enable the enforcement of `objectDefinitions` without a
+name.
+
+## Motivation
+
+The `ConfigurationPolicy` currently has a `namespaceSelector` to configure the controller to filter
+and range over namespaces relevant to a particular namespaced `objectDefinition`. Furthermore,
+currently an `objectDefinition` without a name is feasible but is not enforceable. A
+`resourceSelector` would both pair well with the `namespaceSelector` for further object filtering as
+well as enable an enforceable unnamed `objectDefinition` by providing the controller with a list of
+names to handle.
+
+Adding a `resourceSelector` would lower the bar to entry for `ConfigurationPolicy`, adding an
+additional configuration available to users without requiring knowledge of Go templating as is
+required with the alternative, `object-templates-raw`.
+
+### Goals
+
+- Add a `resourceSelector` to `ConfigurationPolicy` to allow the controller to handle objects
+  without requiring `object-templates-raw`
+- Make unnamed `objectDefinition`s enforceable
+- Make the `namespaceSelector` apply to `Namespace`s when a `resourceSelector` is not provided for a
+  `Namespace` `objectDefinition`, allowing a related `Namespace` to be created alongside related
+  namespaced objects.
+
+### Non-Goals
+
+- Establishing complex logic in the `resourceSelector` that could not be handled readily by the
+  Kubernetes API and, by extension, the `kubernetes-dependency-watches` library.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a policy user, I want to be able to create namespaces based on the `namespaceSelector`.
+
+#### Story 2
+
+As a policy user, I want to be able to select objects by label and have the controller iterate over
+them, both to `inform` and `enforce`.
+
+### Implementation Details/Notes/Constraints
+
+Since the `resourceSelector` refers to a specific object, the `resourceSelector` would be embedded
+in each element of the `object-templates` array. The `resourceSelector` would ONLY be composed of a
+`LabelSelector` since that can be handled by the Kubernetes API. The `namespaceSelector` has
+`include` and `exclude` filters for filtering by name, but such a feature requires that it run in
+its own `Namespace` reconciler, which would be unwieldy with a `resourceSelector` where the `Kind`
+is arbitrary.
+
+In a YAML manifest, this new field would appear as follows:
+
+```yaml
+apiVersion: policy.open-cluster-management.io/v1
+kind: ConfigurationPolicy
+metadata:
+  name: resource-selector-policy
+spec:
+  remediationAction: inform
+  namespaceSelector:        # <---- Existing namespaceSelector
+    exclude: []             #
+    include: []             #
+    matchLabels: {}         #
+    matchExpressions: []    #
+  object-templates:
+    - complianceType: musthave
+      resourceSelector:         # <---- New resourceSelector
+        matchLabels: {}         #
+        matchExpressions: []    #
+      objectDefinition:
```

Comment:
Could you touch on how you see templating playing into this? It'd be cool if for every namespace and resource selector result, you could execute the templates with the input namespace and resource name as template variables. You get this with the object-templates-raw alternative, and this could be a simpler way to get the same effect.